### PR TITLE
Conditionals tests && Proposal new switch translation

### PR DESCRIPTION
**Pull request by lolleko**

### CODE REVIEW

File: test/integration/lua/conditionals.spec.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            return result;`
+            , util.dummyTypes.Number
+        );
+
+        // Execute
+        let result = util.executeLua(lua);
+
+        // Assert
+        Expect(result).toBe(expected);
+    }
```
Comment (Perryvw) [inline test/integration/lua/conditionals.spec.ts]: Could you add a testcase where fallthrough has a visible effect? Like replacing the = with += here? Would like to test the return for inp 2 in this case. (should be 5)

File: src/Transpiler.ts [hunk 2]
```diff
@@ -347,37 +347,59 @@ export class LuaTranspiler {
 
         let result = this.indent + "-------Switch statement start-------\n";
 
+        let jumpTableName = "____switch" + ++this.genVarCounter
```
Comment (Perryvw) [inline src/Transpiler.ts]: Don't inline assignments (incrementor in this case).

File: src/Transpiler.ts [hunk 3]
```diff
@@ -347,37 +347,59 @@ export class LuaTranspiler {
...
-            if (index < clauses.length - 1) {
-                result += this.indent + `goto switchCase${this.genVarCounter + index + 1}\n`;
+            let i = index + 1;
+            if (i < clauses.length && !tsEx.containsStatement(clause.statements, ts.SyntaxKind.BreakStatement)) {
+                let nextClause = clauses[i];
+                while(i < clauses.length
+                    && ts.isCaseClause(nextClause)
+                    && nextClause.statements.length === 0
+                ) {
+                    nextClause = clauses[++i];
```
Comment (Perryvw) [inline src/Transpiler.ts]: Don't inline assignments (incrementor).

File: test/integration/lua/conditionals.spec.ts (hunk 1 above)
Comment (lolleko) [reply test/integration/lua/conditionals.spec.ts]: Added in b5c1582

**Pull request merged**